detox/android/detox/src/main/java/com/wix/detox/espresso/scroll/SwipeHelper.kt

```diff
@@ -0,0 +1,114 @@
+package com.wix.detox.espresso.scroll
+
+import android.view.View
+import androidx.test.espresso.ViewAction
+import androidx.test.espresso.action.*
+import com.wix.detox.common.DetoxErrors
+import com.wix.detox.espresso.common.annot.*
+import com.wix.detox.espresso.utils.AgnosticPoint2D
+import kotlin.math.min
+import kotlin.math.max
+
+private const val EDGE_FUZZ_FACTOR = 0.083f
+
+typealias CreateSwipeAction = (
+        swiper: Swiper,
+        startCoordinatesProvider: CoordinatesProvider,
+        endCoordinatesProvider: CoordinatesProvider,
+        precisionDescriber: PrecisionDescriber
+) -> ViewAction
+
+class SwipeHelper(private val createAction: CreateSwipeAction) {
+
+    fun swipeInDirection(
+            direction: Int,
+            fast: Boolean = true,
+            offset: Double = Double.NaN,
+            startPositionX: Double = Double.NaN,
+            startPositionY: Double = Double.NaN
```

Comment:
The naming here confuses me, still, because it's nevertheless not really a position, but a ratio. Also, best if it stays consistent with the caller's `startOffsetPercentX` and `startOffsetPercentY`.

Reply:
normalizedStartOffsetX, normalizedStartOffsetY probably